src/System.Private.CoreLib/shared/System/Range.cs

```diff
@@ -0,0 +1,46 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System
+{
+    public readonly struct Range : IEquatable<Range>
+    {
+        public Index Start { get; }
+        public Index End { get; }
+
+        private Range(Index start, Index end)
+        {
+            Start = start;
+            End = end;
+        }
+
+        public override bool Equals(object value)
+        {
+            if (value is Range)
+            {
+                Range r = (Range)value;
+                return r.Start.Equals(Start) && r.End.Equals(End);
+            }
+
+            return false;
+        }
```

Comment:
Now that you've added the strongly typed Equals, this can just be:
```C#
public override bool Equals(object value) => value is Range && Equals((Range)value);
```
but don't block your PR on it.